### PR TITLE
fix(cli): correct download-output command displaying resolved UNC path

### PR DESCRIPTION
**Pull request by godobyte**
### What was the problem/requirement? (What/Why)
The `download-output` CLI command can output resolved UNC root paths for mapped drive, leading to confusing logs like below
```
PS C:\Users\yuanbian\Git\deadline\deadline-cloud> deadline job download-output --job-id job-f5c23912e3b946799f9f18d30765847c
Downloading output from Job 'blender-3.5-splash'

Summary of files to download:
    S:\Blender\Output (2 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] \\DESKTOP-XXXXX\SD\Blender
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: 0
> Please enter the new root directory path, or press Enter to keep it unchanged [\\DESKTOP-4LN5768\SD\Blender]: S:\Blender\Output

Summary of files to download:
    \\DESKTOP-XXXXX\SD\Blender\Output\Output (2 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] \\DESKTOP-XXXXX\SD\Blender\Output
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: 0
```

### What was the solution? (How)
Remove path resolve that output UNC path for mapped network drive.

### What is the impact of this change?
The download-output command shows unresolved path for mapped drive.

```
PS C:\Users\yuanbian\Git\deadline\deadline-cloud> hatch run default:deadline job download-output --job-id job-f5c23912e3b946799f9f18d30765847c
Downloading output from Job 'blender-3.5-splash'

Summary of files to download:
    S:\Blender\Output (2 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] S:\Blender
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: 0
> Please enter the new root directory path, or press Enter to keep it unchanged [S:\Blender]: S:\Blender\Out

Summary of files to download:
    S:\Blender\Out\Output (2 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] S:\Blender\Out
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: y
```

### How was this change tested?
- All unit tests passed successfully.
- Done some manual end-to-end tests including cross-platform scenarios, confirming that it worked as expected. 
  - Fix Test
    - Downloading outputs from job with mapped drive on Windows: shows mapped drive letter (used to be UNC format path)
  - Regression Test
    - Downloading outputs from job with UNC format root paths on Windows: shows UNC format root paths
    - Downloading outputs from job with C drive root paths on windows: shows C drive root paths
    - Downloading outputs from job with UNC format root paths on Linux: prompt OS mismatch and ask user to select local path
    - Downloading outputs from job with C drive root paths on Linux: prompt OS mismatch and ask user to select local path
    - Downloading outputs from job with Linux root paths on Linux: download as expected

### Was this change documented?
No.

### Is this a breaking change?
No.

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/deadline/client/cli/_groups/job_group.py [hunk 1]
```diff
@@ -348,28 +347,25 @@ def _download_job_output(
                     new_root = click.prompt(
```
Comment (marofke) [inline src/deadline/client/cli/_groups/job_group.py:347]: I'm not sure how this worked before, but what happens if the user types in a relative path? Ex. `./Downloads`, or something like `~/Desktop` (I'm not sure if Path.resolve() expands `~`, but I imagine we were using `resolve()` before to handle relative paths...)

Comment (godobyte) [reply src/deadline/client/cli/_groups/job_group.py:347]: Good call! I checked those cases, apparently if the supplied path is relative, the CLI will use the current path as root. If the path is root, the CLI will use the given root path.

### Linux
Before
```
⋊> ~/w/b/deadline-cloud on yuanbian/unc-path-resolve-fix ↑ deadline job download-output --job-id job-a36d92bd2e544812933f274f628e02fc          14:26:42
Downloading output from Job 'blender-3.5-splash'

Summary of files to download:
    /Users/yuanbian/thinkbox/Blender (10 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] /Users/yuanbian/thinkbox/Blender
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: 0
> Please enter the new root directory path, or press Enter to keep it unchanged [/Users/yuanbian/thinkbox/Blender]: ./Downloads/

Summary of files to download:
    /Users/yuanbian/workplace/bealine-clients/deadline-cloud/Downloads (10 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] /Users/yuanbian/workplace/bealine-clients/deadline-cloud/Downloads
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: n
```
After
```
⋊> ~/w/b/deadline-cloud on yuanbian/unc-path-resolve-fix ↑ hatch run default:deadline job download-output --job-id job-a36d92bd2e544812933f274f628e02fc
Downloading output from Job 'blender-3.5-splash'

Summary of files to download:
    /Users/yuanbian/thinkbox/Blender (10 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] /Users/yuanbian/thinkbox/Blender
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: 0
> Please enter the new root directory path, or press Enter to keep it unchanged [/Users/yuanbian/thinkbox/Blender]: ./Downloads/

Summary of files to download:
    /Users/yuanbian/workplace/bealine-clients/deadline-cloud/Downloads (10 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] /Users/yuanbian/workplace/bealine-clients/deadline-cloud/Downloads
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: n
```

### Windows
Before
```


PS C:\Users\yuanbian\Git\deadline\deadline-cloud> deadline job download-output --job-id job-f5c23912e3b946799f9f18d30765847c
Downloading output from Job 'blender-3.5-splash'

Summary of files to download:
    S:\Blender\Output (2 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] \\DESKTOP-4LN5768\SD\Blender
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: 0
> Please enter the new root directory path, or press Enter to keep it unchanged [\\DESKTOP-4LN5768\SD\Blender]: .\Download

Summary of files to download:
    C:\Users\yuanbian\Git\deadline\deadline-cloud\Download\Output (2 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] C:\Users\yuanbian\Git\deadline\deadline-cloud\Download
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: n
Output download canceled.
```
After
```
PS C:\Users\yuanbian\Git\deadline\deadline-cloud> hatch run default:deadline job download-output --job-id job-f5c23912e3b946799f9f18d30765847c
Downloading output from Job 'blender-3.5-splash'

Summary of files to download:
    S:\Blender\Output (2 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] S:\Blender
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: 0
> Please enter the new root directory path, or press Enter to keep it unchanged [S:\Blender].\Download 

Summary of files to download:
    C:\Users\yuanbian\Git\deadline\deadline-cloud\Download\Output (2 files)

You are about to download files which may come from multiple root directories. Here are a list of the current root directories:
[0] C:\Users\yuanbian\Git\deadline\deadline-cloud\Download
> Please enter the index of root directory to edit, y to proceed without changes, or n to cancel the download (0, y, n) [y]: n
Output download canceled.
```

Comment (marofke) [reply src/deadline/client/cli/_groups/job_group.py:347]: Awesome, thanks for doing the extra testing!

**Review by marofke — APPROVED** [review]
Great testing coverage Godot, looks good to me!

:ship: :shark:

**Review by chocecil — COMMENTED** [review]
- Can you update the commit messages according to https://www.conventionalcommits.org/en/v1.0.0/#summary?
- No unit tests changed? 
- Do we plan to add corresponding integ tests?

**Comment by godobyte** [comment]
Thanks for reviewing the PR @chocecil !

> * Can you update the commit messages according to https://www.conventionalcommits.org/en/v1.0.0/#summary?

That's good to know, I'll be careful with commit message and description at the time of squash merging.

> * No unit tests changed?

All existing unit tests are passing as expected.

> * Do we plan to add corresponding integ tests?

Integ tests might be a good fit to validate this change since it requires special drive setup in windows, but we should definitely have integ tests for the workflow to make sure there is no regression.

**Comment by chocecil** [comment]
> Thanks for reviewing the PR @chocecil !
> 
> > * Can you update the commit messages according to https://www.conventionalcommits.org/en/v1.0.0/#summary?
> 
> That's good to know, I'll be careful with commit message and description at the time of squash merging.
> 
> > * No unit tests changed?
> 
> All existing unit tests are passing as expected.
> 
> > * Do we plan to add corresponding integ tests?
> 
> Integ tests might be a good fit to validate this change since it requires special drive setup in windows, but we should definitely have integ tests for the workflow to make sure there is no regression.

I mean we changed the code, but how come no unit tests needs to update?

**Comment by godobyte** [comment]
> I mean we changed the code, but how come no unit tests needs to update?

@chocecil If you mean by no added unit tests to validate, that's because this is an edge case on windows with mapped network drive. I'm not sure how we can mock the drive setup in unit test. Without that part, the unit test will just becomes a normal case on windows such as any other lettered drive, which is covered already. I can create a separate task to see what would be the best test strategy to cover this edge case.

**Review by chocecil — APPROVED** [review]
